source/common/upstream/upstream_impl.cc

```diff
@@ -960,6 +960,44 @@ createOptions(const envoy::config::cluster::v3::Cluster& config,
       config.has_http2_protocol_options(), validation_visitor);
 }
 
+LBPolicyConfig::LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config) {
+
+  switch (config.lb_policy()) {
+    PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+  case envoy::config::cluster::v3::Cluster::ROUND_ROBIN:
+    if (config.has_round_robin_lb_config()) {
+      lbPolicy_ = std::make_unique<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>(
+          config.round_robin_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::LEAST_REQUEST:
+    if (config.has_least_request_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::LeastRequestLbConfig>(
+          config.least_request_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::RING_HASH:
+    if (config.has_ring_hash_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::RingHashLbConfig>(
+          config.ring_hash_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::MAGLEV:
+    if (config.has_maglev_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::MaglevLbConfig>(
+          config.maglev_lb_config());
+    }
+    break;
+  default:
+    break;
```

Comment:
I don't think default is needed (should be taken care of the PANIC at the beginning).

Reply:
I put in the default because without it, I get a build error saying it also wants a case for Random, Cluster Provided and LB Policy Config. Should I just add these cases and leave them blank?